### DISCUSSION TITLE
utils/video/

### DISCUSSION THREAD

**Discussion by giscus** [roboflow/supervision#1278] [Docs]
# utils/video/

A set of easy-to-use utilities that will come in handy in any computer vision project.

https://supervision.roboflow.com/utils/video/

**Comment by ghost** [comment]
I am facing an issue when saving a new video using the information:

with sv.VideoSink(target_path='target.mp4', video_info=video_info) as sink:
    for frame in frames_generator:
        sink.write_frame(frame=frame)

It saves with frame loss:
For example, my original video is 1:01, but the video generated by the supervision is 1:03.

**Reply by LinasKo** [reply]
This could be due to the `codec` being used, and may be tricky to fix, and I can only help in a limited capacity.

You need to find what codec the file was encoded with. You either need to do some googling, or perhaps this will help: https://www.reddit.com/r/mkvtoolnix/comments/13dswo7/viewing_video_codec_in_windows_explorer/

Then, you can set the `codec` argument for `VideoSink` to be one of [these](https://softron.zendesk.com/hc/en-us/articles/207695697-List-of-FourCC-codes-for-video-codecs).

The unfortunate effect is that it will either work, or it will fail silently. We've chosen the default codec in `supervision` in a way where it's guaranteed to work, but may not be ideal for all input videos, just like you're seeing now.
